pkg/controller/job/job_controller.go

```diff
@@ -1263,23 +1267,23 @@ func pastBackoffLimitOnFailure(job *batch.Job, pods []*v1.Pod) bool {
 // pastActiveDeadline checks if job has ActiveDeadlineSeconds field set and if
 // it is exceeded. If the job is currently suspended, the function will always
 // return false.
-func pastActiveDeadline(job *batch.Job) bool {
+func pastActiveDeadline(job *batch.Job, clock clock.Clock) bool {
 	if job.Spec.ActiveDeadlineSeconds == nil || job.Status.StartTime == nil || jobSuspended(job) {
 		return false
 	}
-	now := metav1.Now()
+	now := metav1.NewTime(clock.Now())
 	start := job.Status.StartTime.Time
 	duration := now.Time.Sub(start)
 	allowedDuration := time.Duration(*job.Spec.ActiveDeadlineSeconds) * time.Second
 	return duration >= allowedDuration
 }
 
-func newCondition(conditionType batch.JobConditionType, status v1.ConditionStatus, reason, message string) *batch.JobCondition {
+func newCondition(conditionType batch.JobConditionType, status v1.ConditionStatus, reason, message string, clock clock.Clock) *batch.JobCondition {
```

Comment:
It actually makes more sense to pass `now` here than the clock.

Reply:
This and `ensureJobConditionStatus` are a bit of a problematic one to refactor. They need to go hand in hand to be able to move them into the `jm`. Which one would you prefer? Sending `now` instead of the clock to both or moving them both into `jm` together ?

Reply:
DOne